packages/vite-plugin-docs/docs/common/ImageCodeBlock.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import CodeBlock from '@theme/CodeBlock';
+
+// Helper function to dedent the code
+const dedent = (str) => {
+  const lines = str.split('\n');
+  const minIndent = Math.min(
+    ...lines.filter(line => line.trim()).map(line => line.match(/^ */)[0].length)
+  );
+  return lines.map(line => line.slice(minIndent)).join('\n');
```

Comment:
```suggestion
  return lines.map(line => line.slice(minIndent)).join('\n').trim();
```
Should we trim the returned code? Looks like there's an extra newline on the top.

<img width="542" alt="image" src="https://github.com/user-attachments/assets/e53d7529-77e4-4823-97b4-375267ecd86c">